File: src/main/java/com/horizex/customer/CustomerApi.java
package com.horizex.customer;

import com.horizex.customer.domain.CustomerService;
import org.springframework.stereotype.Component;

import java.util.Optional;
import java.util.UUID;

@Component
public class CustomerApi {
    private final CustomerService customerService;

    public CustomerApi(CustomerService customerService) {
        this.customerService = customerService;
    }

    public Optional<CustomerDto> findCustomerById(UUID id) {
        return customerService.findCustomerById(id);
    }
}

File: src/main/java/com/horizex/customer/CustomerApi.kt
package com.horizex.customer

import com.horizex.customer.domain.CustomerService
import lombok.extern.slf4j.Slf4j
import org.springframework.stereotype.Service
import java.util.*

@Slf4j
@Service
class CustomerApi(
    private val customerService: CustomerService,
    private val customerDtoMapper: CustomerDtoMapper
) {
    fun getCustomer(customerId: UUID): CustomerDto? {
        return customerService.findCustomerById(customerId).map(customerDtoMapper::toCustomerDto)
            .orElse(null)
    }
}
